Fix some comments in tbprobe.cpp

PedanticHacker opened the pull request:
Fixed some comments. I don't care if you ignore this PR, I was bored.

Disservin commented on src/syzygy/tbprobe.cpp:
```diff
@@ -1036,13 +1045,13 @@ uint8_t* set_sizes(PairsData* d, uint8_t* data) {
     // See https://en.wikipedia.org/wiki/Huffman_coding
     // The canonical code is ordered such that longer symbols (in terms of
-    // the number of bits of their Huffman code) have a lower numeric value,
-    // so that d->lowestSym[i] >= d->lowestSym[i+1] (when read as LittleEndian).
```
`LittleEndian` is the enum name.

PedanticHacker replied:
Yes, my 2nd commit fixes this.

Disservin commented:
@FauziAkram do you mind reading this? You and @PedanticHacker are arguably the ones who change the comments the most, and having multiple comment change commits every month is a bit annoying. So limiting these would be nice... maybe you two can have a look at everything and fix any typos which you spot and after that, we try to leave the comments alone, unless the comment is fundamentally incorrect and not just a typo.

FauziAkram commented on src/syzygy/tbprobe.cpp:
```diff
@@ -605,15 +609,15 @@ int decompress_pairs(PairsData* d, uint64_t idx) {
 
         // If our offset is within the number of values represented by symbol sym,
-        // we are done.
+        // we are done...
```
Why use an ellipsis here?

FauziAkram commented:
I've reviewed the changes, and to be honest, I find them to be legit.

My approach involved modifying only instances where I'm certain my version improves upon the original. While, PedanticHacker took a meticulous approach, delving into the details of every punctuation, denotation, and more.
From my side, I think that I already fixed all the comments that from my point of view needed to be fixed.

PedanticHacker commented:
Please wait, I'm still doing modifications to this file, taking into account everything you have alerted me about. Stay tuned, another commit will be pushed to this PR some time today.

PedanticHacker commented:
Please check if the comment changes are okay now.

PedanticHacker commented:
Since the tbprobe.h file corresponds to the tbprobe.cpp file, I have now also fixed the tbprobe.h file and added the commit to this PR. I think it is appropriate.

mstembera commented on src/syzygy/tbprobe.h:
```diff
@@ -30,19 +30,19 @@ class Position;
 namespace Stockfish::Tablebases {
 
 enum WDLScore {
```
I think the below are better and more concise as they are.  Comments are not necessarily meant to be full sentences.

mstembera commented on src/syzygy/tbprobe.h:
```diff
@@ -30,19 +30,19 @@ class Position;
+    FAIL              = 0,   // The probe has failed (missing a table file)
+    OK                = 1,   // The probe was successful
+    CHANGE_STM        = -1,  // The DTZ should check the other playing side
+    ZEROING_BEST_MOVE = 2    // The best move zeroes the DTZ (a capture or a pawn move)
```
I think the above (Line 33 to 45) are better and more concise as they are. Comments are not necessarily meant to be full sentences.

PedanticHacker replied:
These comment changes in tbprobe.h were done in contrast to the corresponding comments in tbprobe.cpp.

What I find inconsistent throughout the whole Stockfish codebase is the fact that some comments are full sentences, while others are not. There is no rule at all as to when a comment needs to be a full sentence and when a concise sentence.

I propose to either have full-sentence comments everywhere in the codebase or have concise ones everywhere. But since the long comments we have (spanning multiple lines) can't really be written in concise form, we should write them in the full-sentence form everywhere.

Maybe you think `meh, I don't care`, but comment form is also important in source code.

Disservin replied:
Stockfish source code doesn’t have to be a novel. Having to write full sentence comments everywhere is way too tedious and annoying.

PedanticHacker replied:
Comments without the `the`, `a` and `an` read very much like broken English.

vdbergh commented:
Comment style is a subject of endless debate. Since there is no "best style", and a fixed style cannot be enforced programmatically, my feeling is that the choice should be left (within reason) to the author of the code. 

IMHO the only changes that are legitimate are 
- corrections of glaring typos;
- clarifications.

PedanticHacker commented:
As I said, you can ignore this PR (i.e., close it) if you feel that it doesn't improve Stockfish comments in any way.
